### PR TITLE
View Client Profile in a separate window, Update UG, DG

### LINKED ISSUE #30: As a forgetful trainer, I can view my clients' contact information
So that I can contact my clients' when I want to update them about certain information

### PR DESCRIPTION
**Pull request by tanweijie123**
Fixes #30 

New Command: cview [INDEX] 
- Ability to view the full client information
- Able to extend for future requirements (eg. weight tracking, payment history, etc.) 

New Ui FXML
- ClientInfoPage.fxml [View] 
- ClientInfoPage.java [Controller] 

Updated UG, DG to reflect changes to cview, cfind.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -315,6 +319,55 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
       Use case resumes at step 2.
 
+---
+
+**Use case: Find Clients**
+
+**MSS**
+
+1.  User requests to find some Client based on keyword or text. 
+2.  AddressBook displays the client's whose name matches the keyword or text.
```
Comment (benclmnt) [inline docs/DeveloperGuide.md]:
```suggestion
2.  FitEgo displays the client's whose name matches the keyword or text. 
```

I see similar issues in other place as well

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 2]
```diff
@@ -45,6 +48,16 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
...
+        /* DISPLAY ANY OTHER UIs */
+        //ViewClientCommand has an additional UI to display profile
+        if (command instanceof ViewClientCommand) {
+            ViewClientCommand viewClient = (ViewClientCommand) command;
+            Stage clientView = new Stage();
+            ClientInfoPage cip = new ClientInfoPage(viewClient.getClient());
+            cip.start(clientView);
+            cip.show();
+        }
+
```
Comment (benclmnt) [inline src/main/java/seedu/address/logic/LogicManager.java]: Perhaps its better to move this to `ViewClientCommand::execute`?
as to maintain separation between `Ui` and `Logic`  components?

**Review by benclmnt — CHANGES_REQUESTED** [review]
Overall looks nice!

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 2 above)
Comment (tanweijie123) [reply src/main/java/seedu/address/logic/LogicManager.java]: Yes. I agree with you, but UI cannot run test cases. 
Putting it in ViewClientCommand cannot test execute. 

Usually, it should be placed in MainWindow::executeCommand, but I will see how
Edit: ViewClientCommand will create a lambda runnable for MainWindow to run.

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -182,6 +182,10 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleHelp();
             }
 
+            if (commandResult.anyFunctionToRun()) {
```
Comment (dhafinrazaq) [inline src/main/java/seedu/address/ui/MainWindow.java]: should the name be something like "hasFunctionToRun"?

**Review by dhafinrazaq — COMMENTED** [review]
Comment on coding standard

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (benclmnt) [reply src/main/java/seedu/address/ui/MainWindow.java]: I agree with Dhafin.

i think in a javascript world, it is called a callback. 
Perhaps `hasCallback` and `getCallback` is a better name?

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 2 above)
Comment (benclmnt) [reply src/main/java/seedu/address/logic/LogicManager.java]: Sounds good!

**Review by benclmnt — APPROVED** [review]
Overall, lgtm!

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (tanweijie123) [reply src/main/java/seedu/address/ui/MainWindow.java]: Updated

**Pull request merged**